pkg/systemstatsmonitor/memory_collector_linux.go

```diff
@@ -38,30 +38,30 @@ func (mc *memoryCollector) collect() {
 	}
 
 	if mc.mBytesUsed != nil {
-		memUsed := meminfo.MemTotal - meminfo.MemFree - meminfo.Buffers - meminfo.Cached - meminfo.Slab
-		mc.mBytesUsed.Record(map[string]string{stateLabel: "free"}, int64(meminfo.MemFree)*1024)
+		memUsed := *meminfo.MemTotal - *meminfo.MemFree - *meminfo.Buffers - *meminfo.Cached - *meminfo.Slab
```

Comment:
Do you think if we need to make sure pointer is not nil.

Reply:
It's a good idea, I've added a check to the condition.

Reply:
I mean maybe we should check if meminfo.MemFree, meminfo.Buffers or other item is nil

Reply:
Gotcha, I updated the condition. Does this fit what you meant?